### PR TITLE
Add HEAD_BODY attribute

### PR DESCRIPTION
**Pull request by yawkat**
For jax-rs tck

### CODE REVIEW

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java [hunk 1]
```diff
@@ -564,6 +564,7 @@ private MutableHttpResponse<?> finaliseResponse(HttpRequest<?> request, RouteInf
                 referenceCounted.release();
             }
             response.body(null);
+            response.attribute(HttpAttributes.HEAD_BODY, o);
```
Comment (dstepanov) [inline http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:567]: Isn’t it a bit strange to add a released body?

**Comment by dstepanov** [comment]
What spec defines this behavior?

**Review by graemerocher — APPROVED** [review]

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java (hunk 1 above)
Comment (yawkat) [reply http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:567]: We only need it in jax-rs for JaxRsResponse which is not ref counted

**Comment by yawkat** [comment]
@dstepanov it's necessary for JaxRsResponseFilter. the body will not actually be sent, but JaxRsResponseFilter needs access to set e.g. headers

**Review by dstepanov — APPROVED** [review]

**Pull request merged**